src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp

```diff
@@ -431,17 +431,23 @@ struct deconvolution : public primitive_base<deconvolution> {
         ib >> grouped_weights_shape;
         ib >> output_partial_shape;
         ib >> output_shape_id;
-        ib >> *const_cast<primitive_id_arr*>(&weights);
-        ib >> *const_cast<primitive_id_arr*>(&bias);
+        ib >> weights;
+        ib >> bias;
     }
 
 protected:
-    std::vector<input_info> get_dependencies() const override {
-        std::vector<input_info> ret;
-        ret.reserve(weights.size() + bias.size() + (output_shape_id.empty() ? 0 : 1));
-        for (auto& w : weights) ret.push_back(w);
-        for (auto& b : bias) ret.push_back(b);
-        if (!output_shape_id.empty()) ret.push_back(output_shape_id);
+    std::map<size_t, const input_info*> get_dependencies_map() const override {
+        auto ret = std::map<size_t, const input_info*>{};
+        auto idx = input.size();
+
+        if (weights.is_valid())
```

Comment:
nit: what about `OPENVINO_ASSERT(weights.is_valid())`?

Reply:
Updated for Deconv/Conv/FC, thanks!